Re-added namequerytype and multilingual parameters to search API

justinjnlim opened the pull request:
Allowing parameters for multilingual name matching and whole vs partial name matching to query string.

justinjnlim committed:
Re-added namequerytype and multilingual parameters to search API

Allowing parameters for multilingual name matching and whole vs partial name matching to query string.

baieric commented on tests/test_functional.py:
```diff
@@ -77,8 +77,13 @@
+    ('get', '/bp/api/search?bpsearch=anglai', None, '[{"message": "No matching languoids found for \'anglai\'"}]'),
+    ('get', '/bp/api/search?bpsearch=anglai&multilingual=true', None, '[{"glottocode": "stan1293", "iso": "eng", "name": "English", "level": "language"}, {"glottocode": "midd1317", "iso": "enm", "name": "Middle English", "level": "language"}, {"glottocode": "tsha1245", "iso": "tsj", "name": "Tshangla", "level": "language"}]'),
+    ('get', '/bp/api/search?bpsearch=klar', None, '[{"glottocode": "kumy1244", "iso": "kum", "name": "Kumyk", "level": "language"}]'),
+    ('get', '/bp/api/search?bpsearch=klar&namequerytype=whole', None, '[{"message": "No matching languoids found for \'klar\'"}]'),
```
Add tests for < 3 characters searched, searching a glottocode, and when there's no results. Also move all these blueprint tests to a separate test function. The function would have the same code as test_pages, just separate for organization's sake.

baieric replied:
take a closer look at the api search function to see if there's any other test coverage we haven't thought of yet. Also we should add a comment for every one of these tests that quickly explains the scenario being tested.

justinjnlim replied:
done!

baieric reviewed (changes requested):
just more organized unit testing needed!

justinjnlim committed:
Added test coverage

Isolated Blueprint tests into test_search_api, adding documentation as well as additional test coverage for the api.

justinjnlim commented:
added/organized the tests!

baieric reviewed (approved):
fantastic, LGTM!
